corehq/apps/locations/expressions.py

```diff
@@ -0,0 +1,35 @@
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.userreports.expressions.factory import ExpressionFactory
+from corehq.apps.userreports.specs import TypeProperty
+from corehq.util.quickcache import quickcache
+from dimagi.ext.jsonobject import JsonObject, DictProperty
+
+
+class LocationTypeSpec(JsonObject):
+    type = TypeProperty('location_type_name')
+    location_id_expression = DictProperty(required=True)
+
+    def __call__(self, item, context=None):
+        location_id_expression = ExpressionFactory.from_spec(self.location_id_expression)
```

Comment:
this should move to a `configure` function or to the `__init__` so that it doesn't get rebootstrapped on every doc. see, e.g. https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/userreports/expressions/specs.py#L166